rush-plugins/rush-resolver-cache-plugin/src/index.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type {
+  IRushPlugin,
+  RushSession,
+  RushConfiguration,
+  IRushCommand,
+  ILogger,
+  Subspace
+} from '@rushstack/rush-sdk';
+
+/**
+ * Plugin that caches information from the package manager after installation to speed up resolution of imports.
+ * @beta
+ */
+export default class RushResolverCachePlugin implements IRushPlugin {
+  public readonly pluginName: 'RushResolverCachePlugin' = 'RushResolverCachePlugin';
+
+  public apply(rushSession: RushSession, rushConfiguration: RushConfiguration): void {
+    rushSession.hooks.afterInstall.tapPromise(
+      this.pluginName,
+      async (command: IRushCommand, subspace: Subspace) => {
+        const logger: ILogger = rushSession.getLogger('RushResolverCachePlugin');
+
+        if (subspace !== rushConfiguration.defaultSubspace) {
+          logger.terminal.writeLine(`Skipping ${this.pluginName} for non-default subspace`);
+          return;
+        }
+
+        if (rushConfiguration.packageManager !== 'pnpm') {
+          logger.emitError(
+            new Error('The RushResolverCachePlugin currently only supports the "pnpm" package manager')
+          );
+          return;
+        }
+
+        if (!rushConfiguration.packageManagerToolVersion.startsWith('8.')) {
+          logger.emitError(new Error('The RushResolverCachePlugin currently only supports pnpm version 8.*'));
```

Comment:
Not pnpm 9?

Reply:
Lockfile format changed again and I haven't picked up the changes.

Reply:
Though in theory it might work? Updated to allow it.